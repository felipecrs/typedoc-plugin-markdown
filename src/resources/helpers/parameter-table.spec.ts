import * as Handlebars from 'handlebars';

import { setup } from '../../../test/utils';

describe(`[helpers] parameterTable`, () => {
  let project: any;

  beforeAll(() => {
    ({ project } = setup());
  });

  test(`should compile`, () => {
    const data = project.findReflectionByName('functionWithDefaults');
    const result = Handlebars.helpers.parameterTable.call(data.signatures[0].parameters);
    expect(result).toMatchSnapshot();
  });
});
