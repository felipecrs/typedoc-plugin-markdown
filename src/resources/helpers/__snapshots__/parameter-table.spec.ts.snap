// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[helpers] parameterTable should compile 1`] = `
"
Name | Type | Default | Description |
------ | ------ | ------ | ------ |
\`valueA\` | string | \\"defaultValue\\" | A parameter with a default string value. |
\`valueB\` | number | 100 | A parameter with a default numeric value. |
\`valueC\` | number | Number.NaN | A parameter with a default NaN value. |
\`valueD\` | boolean | true | A parameter with a default boolean value. |
\`valueE\` | boolean | null | A parameter with a default null value. |
"
`;
